common/tools/eslint-plugin-azure-sdk/src/rules/ts-apiextractor-json-types.ts

```diff
@@ -3,69 +3,84 @@
 
 /**
  * @file Rule to force the inclusion of type declarations in the package.
- * @author Arpan Laha
- * @author Will Temple
  */
 
-import { getRuleMetaData, getVerifiers, stripPath } from "../utils";
-import { Property } from "estree";
-import { Rule } from "eslint";
-import { stripFileName } from "../utils/verifiers";
+import { createRule, getVerifiers, stripPath } from "../utils";
+import { TSESTree } from "@typescript-eslint/utils";
+import { VerifierMessages, stripFileName } from "../utils/verifiers";
 
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
-export = {
-  meta: getRuleMetaData(
-    "ts-apiextractor-json-types",
-    "force api-extractor.json to configure types in a consistent way",
-  ),
-  create: (context: Rule.RuleContext): Rule.RuleListener => {
+export default createRule({
+  name: "ts-apiextractor-json-types",
+  meta: {
+    type: "suggestion",
+    docs: {
+      description: "force api-extractor.json to configure types in a consistent way",
+      recommended: "recommended",
+    },
+    messages: {
+      ...VerifierMessages,
```

Comment:
I am curious: `...VerifierMessages` is needed anytime when we use verifiers methods? is it possible some of the messages are not needed by a rule (which I think is fine)?

Reply:
Yeah the way it works is they type-check the message ids when you call report, so I had to type the verifiers to take a context that had all the messages they might raise when you use them. Unused messages don't do anything as far as I can tell.